src/ray/core_worker/core_worker.cc

```diff
@@ -609,24 +597,62 @@ Status CoreWorker::ExecuteTask(const TaskSpecification &task_spec,
     return_ids.push_back(task_spec.ReturnId(i));
   }
 
-  Status status;
-  TaskType task_type = TaskType::NORMAL_TASK;
-  if (task_spec.IsActorCreationTask()) {
+  TaskType task_type;
+  resource_ids_ = resource_ids;
+  main_thread_task_id_ = task_spec.TaskId();
+  if (task_spec.IsNormalTask()) {
+    RAY_CHECK(current_job_id_.IsNil());
+    current_job_id_ = task_spec.JobId();
+    task_type = TaskType::NORMAL_TASK;
+  } else if (task_spec.IsActorCreationTask()) {
+    RAY_CHECK(current_job_id_.IsNil());
+    current_job_id_ = task_spec.JobId();
     RAY_CHECK(return_ids.size() > 0);
     return_ids.pop_back();
+    RAY_CHECK(actor_id_.IsNil());
+    actor_id_ = task_spec.ActorCreationId();
     task_type = TaskType::ACTOR_CREATION_TASK;
-    SetActorId(task_spec.ActorCreationId());
-  } else if (task_spec.IsActorTask()) {
+  } else {
+    RAY_CHECK(current_job_id_ == task_spec.JobId());
+    RAY_CHECK(actor_id_ == task_spec.ActorId());
```

Comment:
the if-else-if-else block above is getting a bit too complicated. Please consider extract them to a separate function?